passport.js

```diff
@@ -0,0 +1,42 @@
+const passport = require('passport'),
+  LocalStrategy = require('passport-local').Strategy,
+  Models = require('./models.js'),
+  passportJWT = require('passport-jwt');
+
+let Users = Models.User,
+  JWTStrategy = passportJWT.Strategy,
+  ExtractJWT = passportJWT.ExtractJwt;
+
+passport.use(new LocalStrategy({
+  usernameField: 'Username',
+  passwordField: 'Password'
+}, (username, password, callback) => {
+  console.log(username + '  ' + password);
+  Users.findOne({ Username: username }, (error, user) => {
+    if (error) {
+      console.log(error);
+      return callback(error);
+    }
+
+    if (!user) {
+      console.log('incorrect username');
+      return callback(null, false, {message: 'Incorrect username or password.'});
+    }
+
+    console.log('finished');
+    return callback(null, user);
+  });
+}));
+
+passport.use(new JWTStrategy({
+  jwtFromRequest: ExtractJWT.fromAuthHeaderAsBearerToken(),
+  secretOrKey: 'your_jwt_secret'
```

Comment:
this is okay for now but you don't want to expose your secrets on github. This would be better off as an environment variable but we shall learn that later.